### ISSUE TITLE
Question: Export data slow. Am i missing something?

### ISSUE THREAD

**Issue by s-gal** [typesense/typesense#743]
## Description

I have a data-Index with just a bit over 50,000 records. If i just request a small batch to display in Browser (e.g 25) i get results in 0.1 seconds which is very good to me and acceptable.... Even if i have data-Index as 1 million the time stays pretty much the same.

If i try to export the whole data with the export functionality it takes over 70 seconds... Also there is not much CPU used during the export process. (For the index process there is much CPU used).... Is there a trick to increase the export speed?

Also when not using the export function but just using the search endpoint and incrementing the page number everytime i am faster gathering all the data (25 seconds which is still not super fast here)

If i use the multi_search endpoint no mather how i define the MULTI_SEARCH_LIMIT i also be around 25 seconds... 

What possibilities do i have to improve the performance here. Right now i run typesnese self hosted in a docker container. The parameter "thread-pool-size" makes no difference. The parameter Burst as vCPU option seemes only be available for the cloud solution of typesense...

Running out of ideas here. My alternative option would be selecting the data from my postgres database where the data originates and are the base for my typesense data index. But this seemes not to be an attractive way to go, but i would get much better results for 50k records with a SQL than 25 seconds....

Help is much appreciated here...

**Comment by kishorenc** [comment]
@s-gal 

We're probably being a bit too conservative during the export process to prevent searches from being throttled. For now, you can use the snapshot end-point and directly use the on-disk snapshot directory for backup (but this approach won't help if you are looking to process the generated data as JSONL which the export produces).

**Comment by s-gal** [comment]
@kishorenc

Yes i need to be able to apply filters to the data, because i want to build an excel export function which contains all the data the user has filtered (and this of course not just limited then to 25 or 250 results as it shows in their browser but for all results which can be multiple thousand and up to a million of entries)....

So if i understand it right then, typesense is offering no possibility to provide a fast way to export data with a record size of multiple thousands of entries? So this would really something many probably would look forward in a future typesense version, right?

So i probably need to get my data for my excel export for now directly from the database via sql and not from typesense i guess...

**Comment by lionel-ovaert** [comment]
Hi! I see this was moved a while ago in the medium term column. Has this been reviewed recently in terms of priority ? :) I'm currently using the cloud version of Typesense and thus I don't have access to the snapshot API.

**Comment by jasonbosco** [comment]
@lionel-ovaert We did fix a couple of performance issues with the export endpoint since the time the original issue was created. The export endpoint also now supports `filter_by`, `include_fields` and `exclude_fields` (but not the `q` parameter). 

Could you upgrade to the latest version of Typesense and try the export endpoint? If it's still slow, since you're on Typesense Cloud, could you email a curl request of the export command you're using to support at typesense dot org, so we can investigate further?

**Comment by lionel-ovaert** [comment]
@jasonbosco I did give the export endpoint a chance to see whether it had improved since the last time I used it. It seems indeed a bit faster than before. I was able to export 3M records (using a simple filter_by for one field and excluding embedding fields) in ~3 hours. Having quite a powerful machine in Typesense cloud, I did notice however that during the export the CPU did not go above 5% which is a shame as my baseline queries don't consume a lot of CPU in the first place.
